coral-presto/src/main/java/com/linkedin/coral/presto/parser/ParseTreeBuilder.java

```diff
@@ -0,0 +1,1500 @@
+    }
+    operands.add(SqlLiteral.createSample(
+        SqlSampleSpec.createTableSample(node.getType().equals(SampledRelation.Type.BERNOULLI), percentage),
+        getParserPos(node)));
+    return TABLESAMPLE.createCall(getParserPos(node), operands);
+  }
+
+  @Override
+  protected SqlNode visitJoin(Join node, ParserVisitorContext context) {
+    SqlNode left = visitNode(node.getLeft(), context);
+    SqlNode right = visitNode(node.getRight(), context);
+    SqlLiteral natural = createLiteralBoolean(false, ZERO);
+    JoinConditionType conditionType = JoinConditionType.NONE;
+    SqlNode condition = null;
+    if (node.getCriteria().isPresent()) {
+      JoinCriteria joinCriteria = node.getCriteria().get();
+      natural = createLiteralBoolean(joinCriteria instanceof NaturalJoin, ZERO);
+      if (joinCriteria instanceof JoinOn) {
+        conditionType = JoinConditionType.ON;
+        condition = visitNode(((JoinOn) joinCriteria).getExpression(), context);
+      } else if (joinCriteria instanceof JoinUsing) {
+        conditionType = JoinConditionType.USING;
+        condition =
+            createSqlNodeList(getListSqlNode(((JoinUsing) joinCriteria).getColumns(), context), getParserPos(node));
+      }
+    }
+    JoinType joinType = JOIN_TYPE_MAP.get(node.getType());
+    if (joinType == null) {
+      throw getUnsupportedException(node);
+    }
+    return new SqlJoin(getParserPos(node), left, natural, joinType.symbol(ZERO), right, conditionType.symbol(ZERO),
+        condition);
+  }
+
+  @Override
+  protected SqlNode visitExists(ExistsPredicate node, ParserVisitorContext context) {
+    return EXISTS.createCall(getParserPos(node), visitNode(node.getSubquery(), context));
+  }
+
+  private RelDataType getDecimalType(String type) {
+    String value = type.substring(8, type.length() - 1);
+    String[] nums = value.split(",");
+    int precision = Integer.parseInt(nums[0]);
+    int scale = nums.length > 1 ? Integer.parseInt(nums[1]) : 0;
+    return sqlTypeFactory.createSqlType(SqlTypeName.DECIMAL, precision, scale);
+  }
+
+  private RelDataType getVarcharType(String type) {
+    String precision = type.substring(8, type.length() - 1);
+    return sqlTypeFactory.createSqlType(SqlTypeName.VARCHAR, Integer.parseInt(precision));
+  }
+
+  private RelDataType getCharType(String type) {
+    String precision = type.substring(5, type.length() - 1);
+    return sqlTypeFactory.createSqlType(SqlTypeName.CHAR, Integer.parseInt(precision));
+  }
+
+  private RelDataType convertType(String type) {
+    if (type.toUpperCase().startsWith("DECIMAL(")) {
+      return getDecimalType(type);
+    } else if (type.toUpperCase().startsWith("VARCHAR(")) {
+      return getVarcharType(type);
+    } else if (type.toUpperCase().startsWith("CHAR(")) {
+      return getCharType(type);
+    }
+    return sqlTypeFactory.createSqlType(SqlTypeName.valueOf(type.toUpperCase()));
+  }
```

Comment:
Can we leverage `Decimal#getPrecesion` and `Decimal#getScale`, and comparable `VarcharType` methods, etc?